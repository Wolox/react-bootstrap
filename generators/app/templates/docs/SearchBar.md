## SearchBar

**Props**

| Property | Type | Required? | Description |
|:---|:---|:---:|:---|
| submitButtonContent | React.Node | | Button content |
| className | String | | Custom styles to apply to the search bar container |
| buttonClassName | String | | Custom styles to apply to the submit button |
| children | Node | | Elements to be rendered inside the search bar |
| handleSubmit | Func | âœ“ | Function to be called on submit |

**Usage**

```jsx
<SearchBar onSubmit={this.onSubmit}>
  <label for="name" className={styles.labelSearch}>
    Name:
    <input name="name" id="name" type="input" onChange={this.onChange}/>
  </label>
  <label for="lastName" className={styles.labelSearch}>
    Last Name:
    <input name="lastName" id="lastName" type="input" onChange={this.onChange}/>
  </label>
</SearchBar>
```