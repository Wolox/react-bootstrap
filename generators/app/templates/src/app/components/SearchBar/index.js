import React from 'react';
import PropTypes from 'prop-types';

function SearchBarContainer({
  textButtonSearch,
  className,
  formClassName,
  buttonClassName,
  handleSubmit,
  children
}) {
  return (
    <div className={className}>
      <form onSubmit={handleSubmit} className={formClassName}>
        {children}
        <button className={buttonClassName} type="submit">
          {textButtonSearch}
        </button>
      </form>
    </div>
  );
}

SearchBarContainer.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  buttonClassName: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  formClassName: PropTypes.string,
  textButtonSearch: PropTypes.string
};

SearchBarContainer.defaultProps = {
  buttonClassName: '',
  className: '',
  textButtonSearch: 'Buscar'
};

export default SearchBarContainer;
