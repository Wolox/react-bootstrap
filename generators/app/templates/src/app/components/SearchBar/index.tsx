import React from 'react';

interface Props {
  buttonClassName?: string;
  className?: string;
  formClassName?: string;
  submitButtonContent: React.ReactNode;
  children: React.ReactNode;
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}

function SearchBar({ submitButtonContent, className = '', buttonClassName = '', onSubmit, children }: Props) {
  return (
    <form onSubmit={onSubmit} className={className}>
      {children}
      <button className={buttonClassName} type="submit" data-testid="search-bar-submit-button">
        {submitButtonContent}
      </button>
    </form>
  );
}

export default SearchBar;
