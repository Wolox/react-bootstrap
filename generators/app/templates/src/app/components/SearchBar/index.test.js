import React from 'react';
import { mount } from 'enzyme';

import SearchBar from './index';

describe('#SearchBar', () => {
  let component = null;
  const handleSubmit = jest.fn();

  beforeEach(() => {
    component = mount(
      <SearchBar submitButtonContent="Search" onSubmit={handleSubmit}>
        <input />
      </SearchBar>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  describe('when mounting', () => {
    it('matches last snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when pressing the submit button', () => {
    beforeEach(() => {
      const button = component.find('[data-testid="search-bar-submit-button"]');
      button.simulate('submit');
    });

    it('calls the submit function', () => {
      expect(handleSubmit).toHaveBeenCalled();
    });
  });
});
