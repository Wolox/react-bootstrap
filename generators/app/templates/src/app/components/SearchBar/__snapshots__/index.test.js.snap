// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SearchBar when mounting matches last snapshot 1`] = `
<form
  className=""
  onSubmit={[MockFunction]}
>
  <input />
  <button
    className=""
    data-testid="search-bar-submit-button"
    type="submit"
  >
    Search
  </button>
</form>
`;
